osf/models/preprint_service.py

```diff
@@ -74,11 +74,25 @@ def deep_url(self):
 
     @property
     def url(self):
-        return '/{}/'.format(self._id)
+        if self.provider._id == 'osf' or self.provider.domain:
+            return '/{}/'.format(self._id)
+
+        return '/preprints/{}/{}/'.format(self.provider._id, self._id)
+
+    if settings.DEV_MODE:
```

Comment:
FYI, it is much simpler than I thought. No need for checking settings. Staging and Prod will have the same code. Only need to change `local.py` for local development.